Glue client gets 'GlueException' exception when server throws 'EntityNotFoundException'

according to docs 'EntityNotFoundException' should be thrown when trying to fetch an entity but it's not found.

code to reproduce in patch file, here's the output 
```
java.lang.AssertionError: 
Expecting actual throwable to be an instance of:
  software.amazon.awssdk.services.glue.model.EntityNotFoundException
but was:
  software.amazon.awssdk.services.glue.model.GlueException: Service returned HTTP status code 500 (Service: Glue, Status Code: 500, Request ID: null)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleErrorResponse(CombinedResponseHandler.java:125)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handleResponse(CombinedResponseHandler.java:82)
	at software.amazon.awssdk.core.internal.http.CombinedResponseHandler.handle(CombinedResponseHandler.java:60)
	...(107 remaining lines not displayed - this can be changed with Assertions.setMaxStackTraceElementsDisplayed)

	at io.trino.aws.proxy.glue.TestGlueBase.testComplexSerialization(TestGlueBase.java:131)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1597)

```


[reproduce_wrong_exception_on_client_when_server_throws_EntityNotFoundException.patch](https://github.com/user-attachments/files/19441806/reproduce_wrong_exception_on_client_when_server_throws_EntityNotFoundException.patch)

--- a/trino-aws-proxy-glue/pom.xml
+++ b/trino-aws-proxy-glue/pom.xml
@@ -66,6 +66,16 @@
 
         <dependency>
             <groupId>software.amazon.awssdk</groupId>
+            <artifactId>aws-core</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
+            <artifactId>aws-json-protocol</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>software.amazon.awssdk</groupId>
             <artifactId>glue</artifactId>
         </dependency>
 
@@ -112,12 +122,6 @@
         <dependency>
             <groupId>software.amazon.awssdk</groupId>
             <artifactId>auth</artifactId>
-            <scope>runtime</scope>
-        </dependency>
-
-        <dependency>
-            <groupId>software.amazon.awssdk</groupId>
-            <artifactId>aws-core</artifactId>
             <scope>runtime</scope>
         </dependency>
 
@@ -170,4 +174,19 @@
             <scope>test</scope>
         </dependency>
     </dependencies>
+
+    <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <configuration>
+                    <ignoredNonTestScopedDependencies>
+                        <!-- I can't figure out what the dependency plugin is complaining about - punt for now -->
+                        <ignoredDependency>software.amazon.awssdk:aws-core</ignoredDependency>
+                    </ignoredNonTestScopedDependencies>
+                </configuration>
+            </plugin>
+        </plugins>
+    </build>
 </project>

--- a/trino-aws-proxy-glue/src/main/java/io/trino/aws/proxy/glue/rest/TrinoGlueResource.java
+++ b/trino-aws-proxy-glue/src/main/java/io/trino/aws/proxy/glue/rest/TrinoGlueResource.java
@@ -26,14 +26,18 @@ import jakarta.ws.rs.WebApplicationException;
 import jakarta.ws.rs.core.Context;
 import jakarta.ws.rs.core.MediaType;
 import jakarta.ws.rs.core.Response;
+import software.amazon.awssdk.services.glue.model.GlueException;
 import software.amazon.awssdk.services.glue.model.GlueRequest;
 import software.amazon.awssdk.services.glue.model.GlueResponse;
+import software.amazon.awssdk.services.glue.model.InternalServiceException;
 
 import java.io.InputStream;
 
 import static jakarta.ws.rs.core.Response.Status.BAD_REQUEST;
+import static jakarta.ws.rs.core.Response.Status.INTERNAL_SERVER_ERROR;
 import static jakarta.ws.rs.core.Response.Status.NOT_FOUND;
 import static java.util.Objects.requireNonNull;
+import static software.amazon.awssdk.protocols.json.internal.unmarshall.JsonErrorCodeParser.X_AMZN_ERROR_TYPE;
 
 @ResourceSecurity(GlueResourceSecurity.class)
 public class TrinoGlueResource
@@ -78,9 +82,18 @@ public class TrinoGlueResource
         try {
             glueResponse = requestHandler.handleRequest(parsedGlueRequest, signingMetadata, requestLoggingSession);
         }
+        catch (GlueException e) {
+            requestLoggingSession.logException(e);
+
+            return Response.status(BAD_REQUEST)
+                    .header(X_AMZN_ERROR_TYPE, e.getClass().getSimpleName() + ":" + e.getMessage())
+                    .build();
+        }
         catch (Exception e) {
             requestLoggingSession.logException(e);
-            throw e;
+            return Response.status(INTERNAL_SERVER_ERROR)
+                    .header(X_AMZN_ERROR_TYPE, InternalServiceException.class.getSimpleName() + ":" + e.getMessage())
+                    .build();
         }
 
         return Response.ok(glueResponse).build();

--- a/trino-aws-proxy-glue/src/test/java/io/trino/aws/proxy/glue/TestGlueBase.java
+++ b/trino-aws-proxy-glue/src/test/java/io/trino/aws/proxy/glue/TestGlueBase.java
@@ -29,7 +29,9 @@ import software.amazon.awssdk.services.glue.model.CreateTableRequest;
 import software.amazon.awssdk.services.glue.model.Database;
 import software.amazon.awssdk.services.glue.model.DatabaseIdentifier;
 import software.amazon.awssdk.services.glue.model.DatabaseInput;
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
 import software.amazon.awssdk.services.glue.model.FederatedDatabase;
+import software.amazon.awssdk.services.glue.model.GetDatabaseRequest;
 import software.amazon.awssdk.services.glue.model.GetDatabasesRequest;
 import software.amazon.awssdk.services.glue.model.GetDatabasesResponse;
 import software.amazon.awssdk.services.glue.model.GetResourcePoliciesRequest;
@@ -52,6 +54,7 @@ import static io.trino.aws.proxy.glue.TestingGlueRequestHandler.POLICY_A;
 import static io.trino.aws.proxy.glue.TestingGlueRequestHandler.POLICY_B;
 import static java.util.Objects.requireNonNull;
 import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 public abstract class TestGlueBase<T extends TestingGlueContext>
 {
@@ -151,5 +154,16 @@ public abstract class TestGlueBase<T extends TestingGlueContext>
                 .partitionIndexes(PartitionIndex.builder().indexName("i1").keys("k1").build())
                 .transactionId("t1")
                 .build());
+    }
+
+    @Test
+    public void testNotFoundException()
+    {
+        assertThatThrownBy(() -> glueClient.getDatabase(
+                GetDatabaseRequest.builder()
+                        .catalogId("1")
+                        .name("blahblah")
+                        .build()))
+                .isInstanceOf(EntityNotFoundException.class);
     }
 }

--- a/trino-aws-proxy-glue/src/test/java/io/trino/aws/proxy/glue/TestingGlueRequestHandler.java
+++ b/trino-aws-proxy-glue/src/test/java/io/trino/aws/proxy/glue/TestingGlueRequestHandler.java
@@ -23,6 +23,8 @@ import software.amazon.awssdk.services.glue.model.CreateDatabaseRequest;
 import software.amazon.awssdk.services.glue.model.CreateDatabaseResponse;
 import software.amazon.awssdk.services.glue.model.CreateTableRequest;
 import software.amazon.awssdk.services.glue.model.Database;
+import software.amazon.awssdk.services.glue.model.EntityNotFoundException;
+import software.amazon.awssdk.services.glue.model.GetDatabaseRequest;
 import software.amazon.awssdk.services.glue.model.GetDatabasesRequest;
 import software.amazon.awssdk.services.glue.model.GetDatabasesResponse;
 import software.amazon.awssdk.services.glue.model.GetResourcePoliciesRequest;
@@ -80,6 +82,11 @@ public class TestingGlueRequestHandler
         return switch (request.glueRequest()) {
             case GetDatabasesRequest _ -> GetDatabasesResponse.builder()
                     .databaseList(Database.builder().name(DATABASE_1).build(), Database.builder().name(DATABASE_2).build())
+                    .build();
+
+            // TODO: add more cases
+            case GetDatabaseRequest getDatabaseRequest -> throw EntityNotFoundException.builder()
+                    .message("'%s.%s' not found".formatted(getDatabaseRequest.catalogId(), getDatabaseRequest.name()))
                     .build();
 
             case GetResourcePoliciesRequest _ -> GetResourcePoliciesResponse.builder()

